template_test/Performance_template.py

```diff
@@ -0,0 +1,228 @@
+
+        # Benchmark name need to be unique in OCS-CI
+        self.benchmark_name = "<name>"
+        self.uuid = uuid4().hex
+        self.crd_data = {
+            "spec": {
+                # values of test_user and clustername are only 'Place holders'
+                "test_user": "Homer simpson",
+                "clustername": "test_cluster",
+                "elasticsearch": {
+                    "server": config.PERF.get("production_es_server"),
+                    "port": config.PERF.get("production_es_port"),
+                    "url": f"http://{config.PERF.get('production_es_server')}:{config.PERF.get('production_es_port')}",
+                },
+            }
+        }
+        # during development use the dev ES so the data in the Production ES will be clean.
+        if self.dev_mode:
+            self.crd_data["spec"]["elasticsearch"] = {
+                "server": config.PERF.get("dev_es_server"),
+                "port": config.PERF.get("dev_es_port"),
+                "url": f"http://{config.PERF.get('dev_es_server')}:{config.PERF.get('dev_es_port')}",
+            }
+
+        # Any other test setup need to be here
+
+    def teardown(self):
+        """
+        Teardown of the test
+
+        """
+
+        # teardown operation need to be here
+        log.info("Cleanup the environment")
+
+    def run(self):
+        """
+        The actual test function
+
+        """
+
+        # Test procedure need to be here
+
+        # collect all results into internal object parameter, if the test run
+        # more then once (preferred), collect them into a list(s).
+
+    def get_time(self):
+        """
+        Getting the current GMT time in a specific format for the ES report
+
+        Returns:
+            str : current date and time in formatted way
+
+        """
+        return time.strftime("%Y-%m-%dT%H:%M:%SGMT", time.gmtime())
+
+    def init_full_results(self, full_results):
+        """
+        Initialize the full results object which will send to the ES server
+
+        Args:
+            full_results (obj): an empty FIOResultsAnalyse object
+
+        Returns:
+            FIOResultsAnalyse (obj): the input object fill with data
+
+        """
+        for key in self.environment:
+            full_results.add_key(key, self.environment[key])
+
+        # full_results.new_index is the index name in the ES
+        full_results.add_key("index", full_results.new_index)
+
+        # if you need, add here all other data you want to have in the report.
+        return full_results
+
+    # This is not mandatory, but if you want to parametrize the test, this is the way
+    @pytest.mark.parametrize(
+        argnames=["Arg1", "ArgN"],
+        argvalues=[
+            pytest.param(
+                *["Parm1", "ParamN"],
+                marks=[pytest.mark.performance, pytest.mark.polarion_id("Polarion ID")],
+            ),
+        ],
+    )
+    def function_test_name(self, Arg1, ArgN):
```

Comment:
Is this template i.e. How a test suppose to be like ? or we are expecting this to be updated dynamically during execution ?

Reply:
i update the comment, to be more understandable on this part of parametrize.